tests/k8s_lib_injection/test_k8s_manual_inject.py

```diff
@@ -101,7 +101,61 @@ def test_inject_asm_admission_controller(self, test_k8s_instance):
+        Checks the request made to the profiling endpoint (/profiling/v1/input)."""
+        response = requests.get(f"http://localhost:{agent_port}/test/session/requests")
+        for request in response.json():
+            if request["url"].endswith("/profiling/v1/input"):
+                return True
+        return False
+
+    def test_profiling_disabled_by_default(self, test_k8s_instance):
+        logger.info(f"Launching test test_profiling_disabled_by_default")
+        logger.info(
+            f": Weblog: [{test_k8s_instance.k8s_kind_cluster.weblog_port}] Agent: [{test_k8s_instance.k8s_kind_cluster.agent_port}]"
+        )
+        test_k8s_instance.deploy_test_agent()
+        test_k8s_instance.deploy_datadog_cluster_agent()
+        # if profiling is enabled force some profiling data to be sent
+        test_k8s_instance.deploy_weblog_as_pod(
+            env={"DD_PROFILING_UPLOAD_PERIOD": "10", "DD_INTERNAL_PROFILING_LONG_LIVED_THRESHOLD": "1500"}
+        )
+        profiling_request_found = self._check_profiling_request_sent(test_k8s_instance.k8s_kind_cluster.agent_port)
+        assert not profiling_request_found, "Profiling should be disabled by default, but a profiling request was found"
+
+    def test_profiling_admission_controller(self, test_k8s_instance):
+        logger.info(f"Launching test test_profiling_admission_controller")
+        logger.info(
+            f": Weblog: [{test_k8s_instance.k8s_kind_cluster.weblog_port}] Agent: [{test_k8s_instance.k8s_kind_cluster.agent_port}]"
+        )
+        test_k8s_instance.deploy_test_agent()
+        test_k8s_instance.deploy_datadog_cluster_agent(features={"datadog.profiling.enabled": "auto"})
+        test_k8s_instance.deploy_weblog_as_pod(
+            env={"DD_PROFILING_UPLOAD_PERIOD": "10", "DD_INTERNAL_PROFILING_LONG_LIVED_THRESHOLD": "1500"}
+        )
+        profiling_request_found = self._check_profiling_request_sent(test_k8s_instance.k8s_kind_cluster.agent_port)
+        assert profiling_request_found, "No profiling request found"
+
+    def test_profiling_override_cluster_env(self, test_k8s_instance):
+        logger.info(f"Launching test test_profiling_override_cluster_env")
+        logger.info(
+            f": Weblog: [{test_k8s_instance.k8s_kind_cluster.weblog_port}] Agent: [{test_k8s_instance.k8s_kind_cluster.agent_port}]"
+        )
+        cluster_agent_config = {
+            "clusterAgent.env[0].name": "DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED",
+            "clusterAgent.env[0].value": "auto",
+        }
+        test_k8s_instance.deploy_test_agent()
+        test_k8s_instance.deploy_datadog_cluster_agent(features=cluster_agent_config)
+        test_k8s_instance.deploy_weblog_as_pod(
+            env={"DD_PROFILING_UPLOAD_PERIOD": "10", "DD_INTERNAL_PROFILING_LONG_LIVED_THRESHOLD": "1500"}
+        )
+        profiling_request_found = self._check_profiling_request_sent(test_k8s_instance.k8s_kind_cluster.agent_port)
```

Comment:
Is there a delay enforced somehow here? It can take between 12 and 90 seconds here for a profile to actually be sent (12 if the library supports both env vars, 90 if it supports neither. I think only Node.js currently supports `DD_INTERNAL_PROFILING_LONG_LIVED_THRESHOLD`)

Reply:
Ok, I changed the method that check if profiling data is sent. Now, I added a 90 seconds timeout